docs/specs/adx/name-resolution.md

```diff
@@ -29,32 +29,128 @@ async function resolveName(name: string, nameServer?: string) {
 resolveName('bob.com', 'http://localhost:1234') // resolve using my local debug nameserver
 ```
 
-## Resolution
+## Example cases
 
-TODO: deprecate this in favor of XRPC resolution above?
+### Hosting service
 
-Here is the algorithm in pseudo-javascript:
+Consider a scenario where a hosting service is using PLC and is providing the username for the user as a subdomain:
 
-```js
-const WELL_KNOWN_PATH = '/.well-known/adx-did'
+- The username: `alice.pds.com`
+- The DID: `did:plc:12345`
+- The hosting service: `https://pds.com`
 
-function resolveName(name: string) {
-  let origin
-  if (name === 'localhost' || name.endsWith('.localhost')) {
-    origin = 'http://localhost'
-  } else {
-    origin = `https://${name}`
-  }
+At first, all we know is `alice.pds.com`, so we call `todo.adx.resolveName()` on `alice.pds.com`. This tells us the DID.
 
-  const res = fetch(`${origin}${WELL_KNOWN_PATH}`, {
-    headers: { Accept: 'text/plain', Host: name },
-  })
-  
-  assert(typeof res === 'string' && res.startsWith('did:'))
-  return res
-}
+```typescript
+await xrpc.service('https://alice.pds.com').todo.adx.resolveName() // => {did: 'did:plc:12345'}
```

Comment:
wait do you call it on `alice.pds.com` or on `pds.com`?

Reply:
There are two situations that you call `resolveName`:
- A client speaking to their own PDS (client/server API)
- Any node trying to resolve the name (names protocol)

In the client/server case, you call it against your PDS:

```typescript
await xrpc.service('https://my-pds.com').todo.adx.resolveName({name: 'alice.pds.com'})
```

In the names protocol case, you call it against the name's origin:

```
await xrpc.service('https://alice.pds.com').todo.adx.resolveName()
```